### DISCUSSION TITLE
How to make resources from repository accessible in exams/homeworks

### DISCUSSION THREAD

**Discussion by luspi** [inducer/relate#891] [Q&A] [answered]
In various homework problems and quiz/exam questions we want to use resources from the git repository (like images for plots, or some data files). If we include in the yaml file something like `[link](repo:path/to/data)`, then we get a `403 Forbidden` error when trying to access that resource from the flow.

What setting do we need to set to allow access to certain data from the underlying repository?

**Comment by inducer** [comment, marked as answer]
You need a file `.attributes.yml` in the same folder as the file you're looking to make accessible, formatted like this:
```yml
unenrolled:
- "*.png"
- "*.jpeg"
```
For each role (`unenrolled`, `student`, `ta`, `instructor`), this provides glob patterns of files that are accessible. "More privileged" roles inherit the access from the "less privileged" ones, by default, via this mechanism:

https://github.com/inducer/relate/blob/c2b196c910a6229922285062717c308eec606135/course/models.py#L664-L667

(i.e. you can change this by configuring roles)

This is documented here: https://documen.tician.de/relate/content.html#custom-urls

**Reply by luspi** [reply]
Thanks, this worked!
